src/ray/gcs/gcs_server/gcs_job_manager.cc

```diff
@@ -31,61 +31,94 @@ void GcsJobManager::Initialize(const GcsInitData &gcs_init_data) {
-  auto on_done = [this, job_id, request, reply,
+                << ", driver pid = " << mutable_job_table_data.driver_pid();
+
+  auto on_done = [this, job_id, mutable_job_table_data, reply,
                   send_reply_callback](const Status &status) {
     if (!status.ok()) {
       RAY_LOG(ERROR) << "Failed to add job, job id = " << job_id
-                     << ", driver pid = " << request.data().driver_pid();
+                     << ", driver pid = " << mutable_job_table_data.driver_pid();
     } else {
       RAY_CHECK_OK(gcs_pub_sub_->Publish(JOB_CHANNEL, job_id.Hex(),
-                                         request.data().SerializeAsString(), nullptr));
-      if (request.data().config().has_runtime_env()) {
-        runtime_env_manager_.AddURIReference(job_id.Hex(),
-                                             request.data().config().runtime_env());
+                                         mutable_job_table_data.SerializeAsString(),
+                                         nullptr));
+      if (mutable_job_table_data.config().has_runtime_env()) {
+        runtime_env_manager_.AddURIReference(
+            job_id.Hex(), mutable_job_table_data.config().runtime_env());
       }
       RAY_LOG(INFO) << "Finished adding job, job id = " << job_id
-                    << ", driver pid = " << request.data().driver_pid();
-      ray_namespaces_[job_id] = request.data().config().ray_namespace();
+                    << ", driver pid = " << mutable_job_table_data.driver_pid();
+      ray_namespaces_[job_id] = mutable_job_table_data.config().ray_namespace();
     }
     GCS_RPC_SEND_REPLY(send_reply_callback, reply, status);
   };
 
-  Status status = gcs_table_storage_->JobTable().Put(job_id, request.data(), on_done);
+  Status status =
+      gcs_table_storage_->JobTable().Put(job_id, mutable_job_table_data, on_done);
   if (!status.ok()) {
     on_done(status);
   }
 }
 
-void GcsJobManager::HandleMarkJobFinished(const rpc::MarkJobFinishedRequest &request,
-                                          rpc::MarkJobFinishedReply *reply,
-                                          rpc::SendReplyCallback send_reply_callback) {
-  JobID job_id = JobID::FromBinary(request.job_id());
-  RAY_LOG(INFO) << "Marking job state, job id = " << job_id;
-  auto job_table_data =
-      gcs::CreateJobTableData(job_id, /*is_dead*/ true, std::time(nullptr), "", -1);
-  auto on_done = [this, job_id, job_table_data, reply,
-                  send_reply_callback](const Status &status) {
+void GcsJobManager::MarkJobAsFinished(rpc::JobTableData job_table_data,
+                                      std::function<void(Status)> done_callback) {
+  const JobID job_id = JobID::FromBinary(job_table_data.job_id());
+
+  auto time = current_sys_time_ms();
+  job_table_data.set_timestamp(time);
+  job_table_data.set_end_time(time);
+  job_table_data.set_is_dead(true);
+  job_table_data.set_driver_ip_address("");
+  job_table_data.set_driver_pid(-1);
```

Comment:
@wuisawesome , do you remember why you dedided to remove the driver_ip_address and driver_pid?

This information is useful after the fact because it allows me to find where the logs for this job are.

Reply:
> `      gcs::CreateJobTableData(job_id, /*is_dead*/ true, std::time(nullptr), "", -1);`

I was just preserving the existing behavior. (If I had to guess/interpret, I'd say that we want the field to be up to date and a user could do `ray.init(); ray.shutdown(); ray.init()` at it would be theoretically possible to have 2 jobs with the same ip and pid.